### PR TITLE
Strip `Evaluated` from arguments in `functionEvaluator`

### LINKED ISSUE #1373: Builtin evaluator should strip `Evaluated` from arguments.
The builtin evaluator should remove `Evaluated` if it appears on the top of arguments. See this error:

```
Expecting hook 'INT.tdiv' to reduce concrete pattern:
    tdivIntId{}(
        /* builtin: */ \dv{intSort{}}("555"),
        /* evaluated: */ /* builtin: */ \dv{intSort{}}("1080")
    )
```

This can be done once for all builtin functions in [`Kore.Builtin.Builtin.functionEvaluator`](https://github.com/kframework/kore/blob/master/kore/src/Kore/Builtin/Builtin.hs#L309).

### PR DESCRIPTION
**Pull request by andreiburdusa**
---
Fixes #1373 
###### Reviewer checklist

- [ ] Test coverage: `stack test --coverage`
- [ ] Public API documentation: `stack haddock`

### CODE REVIEW

File: kore/src/Kore/Builtin/Builtin.hs [hunk 1]
```diff
@@ -318,8 +318,16 @@ functionEvaluator impl =
             (Attribute.Pattern variable)
             (TermLike variable)
         -> simplifier (AttemptedAxiom variable)
-    evaluator (valid :< app) = impl resultSort applicationChildren
+    evaluator (valid :< app) =
+        impl resultSort
+        $ fmap removeEvaluated applicationChildren
       where
+        removeEvaluated :: TermLike variable -> TermLike variable
```
Comment (ttuegel) [inline kore/src/Kore/Builtin/Builtin.hs]: This is probably useful to add to `Kore.Internal.TermLike`.

File: kore/test/Test/Kore/Builtin/Int.hs [hunk 2]
```diff
@@ -240,6 +245,22 @@ test_abs = testUnary absIntSymbol abs
 test_tdiv :: TestTree
 test_tdiv = testPartialBinary tdivIntSymbol tdiv
 
+test_tdiv_evaluated_arguments :: TestTree
```
Comment (andreiburdusa) [inline kore/test/Test/Kore/Builtin/Int.hs]: Dhis test fails without using `removeEvaluated` in `functionEvaluator`, but it does't display the same error as in #1373 .
I hope it's ok

File: kore/test/Test/Kore/Builtin/Int.hs [hunk 3]
```diff
@@ -240,6 +245,22 @@ test_abs = testUnary absIntSymbol abs
...
+        a <- forAll genInteger
+        b <- forAll genInteger
+        na <- forAll $ Gen.integral (Range.linear 0 5)
+        nb <- forAll $ Gen.integral (Range.linear 0 5)
+        let expect = asPartialPattern $ tdiv a b
+        actual <- evaluateT
+            $ mkApplySymbol tdivIntSymbol [evaluated na a, evaluated nb b]
+        (===) expect actual
+  where
+    name = "qq" <> expectHook tdivIntSymbol <> " with evaluated arguments"
```
Comment (ttuegel) [inline kore/test/Test/Kore/Builtin/Int.hs]:
```suggestion
    name = expectHook tdivIntSymbol <> " with evaluated arguments"
```

File: kore/src/Kore/Internal/TermLike.hs [hunk 4]
```diff
@@ -649,6 +650,14 @@ forceSorts operandSorts children =
         , Pretty.indent 4 (Unparser.arguments children)
         ]
 
+{- | Remove Evaluated if it appears on the top of the Term
```
Comment (ttuegel) [inline kore/src/Kore/Internal/TermLike.hs]:
```suggestion
{- | Remove `Evaluated` if it appears on the top of the `TermLike`.
```

File: kore/test/Test/Kore/Builtin/Int.hs (hunk 2 above)
Comment (ttuegel) [reply kore/test/Test/Kore/Builtin/Int.hs]: Yes, that's expected. That error comes from something higher up.

**Pull request merged**